text/001x-battery_improvements.md

```diff
@@ -0,0 +1,286 @@
+The vast majority of consumers are only interested in the total battery voltage, and either sum the battery cells or get the cumulative voltage.
+By separating the cell voltage reporting into a separate message the new battery status can be much smaller, and the cell voltages need only be sent if the consumer is actually interested.
+
+# Detailed Design
+
+There are three parts to the design:
+1. A more efficient status message
+2. A scalable battery voltage message.
+3. Mechanisms that allow the new messages to seamlessly coexist with the old one along with eventual deprecation of the older message.
+
+## BATTERY_STATUS_V2
+
+The proposed message is:
+
+```xml
+    <message id="???" name="BATTERY_STATUS_V2">
+      <description>Battery dynamic information.
+        This should be streamed (nominally at 1Hz).
+        Static/invariant battery information is sent in SMART_BATTERY_INFO.
+
+        The full-battery charge is current_remaining + current_consumed, iff both values are supplied.
+        The `current_remaining` field should only be sent if it is guaranteed to be near-accurate.
+        Power monitors typically should not send the `current_remaining` field as it can only be accurate if the battery is fully charged when the drone is turned on.
+        (A GCS can use `current_remaining` being invalid as a trigger to notify the user to fully charge the battery before flight).
+      </description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
+      <field type="int16_t" name="temperature" units="cdegC" invalid="INT16_MAX">Temperature of the whole battery pack (not internal electronics). INT16_MAX field not provided.</field>
+      <field type="uint32_t" name="voltage" units="mV" invalid="UINT32_MAX">Battery voltage (total). UINT32_MAX: field not provided.</field>
+      <field type="uint32_t" name="current" units="mA" invalid="UINT32_MAX">Battery current (through all cells/loads). UINT32_MAX: field not provided.</field>
+      <field type="uint32_t" name="current_consumed" units="mAh" invalid="UINT32_MAX">Consumed charge (from full). UINT32_MAX: field not provided. Note: Power modules report the current consumed since they were last turned on (the expectation is that batteries are fully charged before turning on the vehicle).</field>
+      <field type="uint32_t" name="current_remaining" units="mAh" invalid="UINT32_MAX">Remaining charge (until empty). UINT32_MAX: field not provided. Note: Power monitors should not set this value.</field>
+      <field type="uint8_t" name="percent_remaining" units="%" invalid="UINT8_MAX">Remaining battery energy. Values: [0-100], UINT32_MAX: field not provided.</field>
+      <field type="uint32_t" name="battery_status" display="bitmask" enum="MAV_BATTERY_STATUS_FLAGS">Fault, health, and readiness status indications.</field>
+    </message>
+```
+
+`MAV_BATTERY_STATUS_FLAGS` is a new enum that includes faults, charge state, and battery mode:
```

Comment:
@dakejahl [Following our very long conversations on battery status flags](https://github.com/mavlink/rfcs/pull/19#issuecomment-1091085450) this is what I think we (more or less) agreed. Note that I have not attempted to match the numbers/ids yet since I might not have it right. Can you please look at this and confirm you think I have it right. If so, I think I can make this "not draft" and start getting broader feedback.

A few notes:
- Removed `STATUS_FLAG_IN_USE` - don't see a need for it now. We can add later if that changes.
- Removed `MAV_BATTERY_FAULT_DEEP_DISCHARGE` as discussed (can be inferred from other flags)
- Kept MAV_BATTERY_STATUS_FLAGS_FAULT_CELL_FAIL - I think it is useful to report that we know cells have died but it is still safe to fly on this battery. If you can never have a cell fail and still consider the battery safe to use then we should remove (?)
- Kept BAD_BATTERY as discussed.
- The fault for a protection system trigger is called MAV_BATTERY_STATUS_FLAGS_FAULT_PROTECTION_SYSTEM
  - FAULT is in name before the type of fault - e.g.  STATUS_FLAGS_FAULT_PROTECTION_SYSTEM - not sure if would be better having fault at end (so ...STATUS_FLAGS_PROTECTION_SYSTEM_FAULT).
- I assume that faults like over voltage might trigger even if the safety fault does not (i.e. you still get the warning).

It wasn't clear to me if we need status "there isn't a fault but we turned on the protection system anyway". Maybe that is just the STATUS_FLAGS_FAULT_PROTECTION_SYSTEM

Reply:
@dakejahl My inclination was to leave `MAV_BATTERY_STATUS_FLAGS_FAULT_CELL_FAIL` - if you got this and bad battery you would then know the cause of the issue. But we can always add it back later if there is a general feeling it is needed (so I will removed).

PROTECTION discussed in the other comment you mande